serverless-cljs-plugin/serverless_lumo/build.cljs

```diff
@@ -44,13 +44,18 @@
 (defn compile!
   "Compile ClojureScript using the lumo api."
   [inputs compiler-opts]
-  (js/console.info
-   "Invoking the Lumo compiler w/ inputs"
-   (.stringify js/JSON (clj->js inputs)))
-  (run! classpath/add! inputs)
-  (lumo.build.api/build
-   (apply lumo.build.api/inputs inputs)
-   compiler-opts))
+  (js/Promise.
+   (fn [resolve reject]
+     (js/console.info
+      "Invoking the Lumo compiler w/ inputs"
+      (.stringify js/JSON (clj->js inputs)))
+     (run! classpath/add! inputs)
+     (lumo.build.api/build
+      (apply lumo.build.api/inputs inputs)
+      compiler-opts
+      nil
+      #(let [promise-fn (if (:error %) reject resolve)]
+         (promise-fn %))))))
```

Comment:
This would be clearer if the only statement inside the promise was:

```clojure
(js/Promise.
  (fn [resolve reject]
    (lumo.build.api/build
       ...
       (fn [result]
         (if (:error result)
           (reject result)
           (resolve result))))
```

What happens if the user doesn't have the latest lumo?